src/main/java/bio/terra/workspace/service/datareference/flight/GenerateReferenceIdStep.java

```diff
@@ -0,0 +1,27 @@
+package bio.terra.workspace.service.datareference.flight;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class GenerateReferenceIdStep implements Step {
```

Comment:
add javadoc about the purpose of this step and why it's important to do as a separate step. Every public class should have some top level javadoc: https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required